src/jdk.httpserver/share/classes/module-info.java

```diff
@@ -25,6 +25,39 @@
 
 /**
  * Defines the JDK-specific HTTP server API.
+ * <p>
+ * A basic high-level API for building embedded servers. Both HTTP and
+ * HTTPS are supported.
+ * <p>
+ * The main components are:
+ * <ul>
+ * <li>the {@link com.sun.net.httpserver.HttpExchange} class that describes a
+ * request and response pair,</li>
+ * <li>the {@link com.sun.net.httpserver.HttpHandler} interface to handle
+ * incoming requests, plus the {@link com.sun.net.httpserver.HttpHandlers} class
+ * that provides useful handler implementations,</li>
+ * <li>the {@link com.sun.net.httpserver.HttpContext} class that maps a URI path
+ * to a {@code HttpHandler},</li>
+ * <li>the {@link com.sun.net.httpserver.HttpServer} class to listen for
+ * connections and dispatch requests to handlers,</li>
+ * <li>the {@link com.sun.net.httpserver.Filter} class that allows pre- and post-
+ * processing of requests.</li></ul>
+ * <p>
+ * The {@link com.sun.net.httpserver.SimpleFileServer} class offers a simple
+ * HTTP file server (intended for testing, development and debugging purposes
+ * only). A default implementation is provided via the <a id="entry-point"></a>
+ * main entry point of the {@code jdk.httpserver} module, which can be used on
+ * the command line as such:
+ * <pre>{@code
+ *    Usage: java -m jdk.httpserver [-b bind address] [-p port] [-d directory]
+ *                                  [-o none|info|verbose] [-h to show options]
+ *    Options:
+ *    -b, --bind-address    - Address to bind to. Default: 0.0.0.0 (all interfaces).
```

Comment:
We did consider defaulting to the loopback address, but this would limit the usefulness of the server too much in the default configuration as it can only be accessed from localhost. The goal of this JEP is an out-of-the-box web server with easy setup, so in this case we favour usability. The purpose of a web server is to make things accessible on the web so it is assumed that the developer is familiar with the terms this comes with. 

The concern of accidental exposure is alleviated by the informative output printed at start up, e.g. 
```~ $ java-sb -m jdk.httpserver
Serving /current/directory and subdirectories on 0.0.0.0:8000
http://123.456.7.891:8000/ ...
```
Considering your point though, we can spell out all interfaces and describe the URL more clearly:

```~ $ java-sb -m jdk.httpserver
Serving /current/directory and subdirectories on 0.0.0.0 (all interfaces) port 8000
Localhost URL: http://123.456.7.891:8000/ ...
```